Algorithms/Python/insertion_sort.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import TypeVar
+
+T = TypeVar("T", bound=int | float)
+
+def insertion_sort(arr):
```

Comment:
Python don't have any type annotations by default.

```c
// C do have such "annotations"
int var = 1;
```
```py
# but in python you do this
var: int = 1
```

---

```c
// for array
int arr[] = {1, 2, 3, 4, 5};
```
```py
# for list
arr: list[int] = [1, 2, 3, 4, 5]
```

---

So you need to annotate `arr` here with

```py
def insertion_sort(arr: list[T]) -> None:
```
This is similar to:
```c
void insertion_sort(int arr[])
```

If our list is of list of integers, then we annotate like `list[int]`, if list of floats, then `list[float]`. But if list of int or float? We defined a `TypeVar` for it, that can determine the variable is int or float.

```
T <- int or float

arr: list[T]
```